src/vs/platform/actions/common/actions.ts

```diff
@@ -246,6 +246,10 @@ export class MenuItemAction extends ExecuteCommandAction {
 		typeof item.title === 'string' ? super(item.id, item.title, commandService) : super(item.id, item.title.value, commandService);
 		this._cssClass = undefined;
 		this._enabled = !item.precondition || contextKeyService.contextMatchesRules(item.precondition);
+		if (item.effectToggled && contextKeyService.contextMatchesRules(item.effectToggled)) {
+			this._checked = true;
+		}
```

Comment:
better, clearer? `this._check = item.effectToggled && contextKeyService.contextMatchesRules(item.effectToggled)`